packages/core/src/animation/Animator.ts

```diff
@@ -302,7 +302,8 @@ export class Animator extends Component {
 
   private _saveAnimatorEventHandlers(state: AnimatorState, animatorStateData: AnimatorStateData): void {
     const eventHandlerPool = this._animationEventHandlerPool;
-    const scripts = this._entity._scripts;
+    const scripts = [];
+    this._entity.getComponents(Script, scripts);
```

Comment:
this._entity._scripts can only obtain scripts when this._entity.isActiveInHierarchy=true. However, there's a possibility for users to play the animation when isActiveInHierarchy=false, so it's changed to obtain the scripts through getComponents(Script).